libfreerdp/core/utils.c

```diff
@@ -148,6 +148,9 @@ auth_status utils_authenticate(freerdp* instance, rdp_auth_reason reason, BOOL o
 	if (freerdp_shall_disconnect_context(instance->context))
 		return AUTH_FAILED;
 
+	if (settings->ConnectChildSession)
```

Comment:
better use `freerdp_settings_get_bool`

Reply:
I'm not for using `freerdp_settings_get_bool` inside the FreeRDP library, accessing directly the struct member allows the compiler to do some type checks, while with `freerdp_settings_get_bool` we can have errors that we will discover only at runtime (if we ever hit that code path). And also the code is easier to read, which is also a big concern.
But I guess that we must have a discussion on the usage of these functions.

Reply:
well, we can have these checks with `freerdp_settings_get_bool` too (and many more static and runtime checks)
the direct access we have currently prevents implementing these as we never know the state of the fields as it might be changed in between outside the scope of such internal checks